kai-analyzer/main.go

```diff
@@ -54,7 +55,20 @@ func main() {
 	}
 
 	l := logr.FromSlogHandler(logger)
-	l.Info("Starting Analyzer", "source-dir", *sourceDirectory, "rules-dir", *rulesDirectory)
+
+	// Check if Java exists on the PATH
+	if err := exec.Command("java", "-version").Run(); err != nil {
+		panic("Java is not installed or not on the PATH")
+	}
+	l.Info("Java is installed")
+
+	// Check if Maven exists on the PATH
+	if err := exec.Command("mvn", "-version").Run(); err != nil {
+		panic("Maven is not installed or not on the PATH")
+	}
```

Comment:
yay or nay on this? This, while not the ultimate problem with the extension integration, would have been really helpful early on when I hadn't installed these.

Reply:
I like it, this is a good idea.

We even may want to consider logging the versions because they might have some known issues if you are using older Java or Maven.